zipkin-storage/zipkin2_cassandra/src/main/resources/zipkin2-schema.cql

```diff
@@ -0,0 +1,108 @@
+    PRIMARY KEY (trace_id, ts_uuid, id)
+)
+    WITH CLUSTERING ORDER BY (ts_uuid DESC)
+    AND compaction = {'class': 'org.apache.cassandra.db.compaction.TimeWindowCompactionStrategy'}
+    AND default_time_to_live =  604800
+    AND gc_grace_seconds = 3600
+    AND read_repair_chance = 0
+    AND dclocal_read_repair_chance = 0.0
+    AND speculative_retry = '95percentile'
+    AND comment = 'Primary table for holding trace data. annotation_query column is a comma separated string (instead of a set) to permit SASI (CASSANDRA-11182)';
+
+
+CREATE TABLE IF NOT EXISTS zipkin2.trace_by_service_span (
+    service       text,             //-- service name
+    span          text,             //-- span name, or blank for queries without span name
+    bucket        int,              //-- time bucket, calculated as ts/interval (in microseconds), for some pre-configured interval like 1 day.
+    ts            timeuuid,         //-- start timestamp of the span, truncated to millisecond precision
+    trace_id      text,             //-- trace ID
+    duration      bigint,           //-- span duration, in tens of milliseconds (or hundredths of seconds)
+    PRIMARY KEY ((service, span, bucket), ts)
+)
+   WITH CLUSTERING ORDER BY (ts DESC)
+    AND compaction = {'class': 'org.apache.cassandra.db.compaction.TimeWindowCompactionStrategy'}
+    AND default_time_to_live =  259200
+    AND gc_grace_seconds = 3600
+    AND read_repair_chance = 0
+    AND dclocal_read_repair_chance = 0
+    AND speculative_retry = '95percentile'
+    AND comment = 'Secondary table for looking up a trace by a service, or service and span. span column may be blank (when only looking up by service). bucket column adds time bucketing to the partition key, values are microseconds rounded to a pre-configured interval (typically one day). ts column is start timestamp of the span as time-uuid, truncated to millisecond precision. duration column is span duration, rounded up to tens of milliseconds (or hundredths of seconds)';
+
+CREATE TABLE IF NOT EXISTS zipkin2.span_by_service (
+    service text,
+    span    text,
+    PRIMARY KEY (service, span)
+)
+    WITH compaction = {'class': 'org.apache.cassandra.db.compaction.LeveledCompactionStrategy', 'unchecked_tombstone_compaction': 'true', 'tombstone_threshold': '0.2'}
+    AND caching = {'rows_per_partition': 'ALL'}
+    AND default_time_to_live =  259200
+    AND gc_grace_seconds = 3600
+    AND read_repair_chance = 0
+    AND dclocal_read_repair_chance = 0
+    AND speculative_retry = '95percentile'
+    AND comment = 'Secondary table for looking up span names by a service name.';
+
+CREATE CUSTOM INDEX IF NOT EXISTS ON zipkin2.span (annotation_query) USING 'org.apache.cassandra.index.sasi.SASIIndex'
+   WITH OPTIONS = {
+    'mode': 'CONTAINS',
+    'analyzed': 'true',
+    'analyzer_class':'org.apache.cassandra.index.sasi.analyzer.NonTokenizingAnalyzer',
+    'case_sensitive': 'false'
+   };
```

Comment:
tests that CassandraSpanStore still works smoothly after dropping the SASI index would be a fantastic addition.

this SASI index is the limiting factor in being able to truly scale this backend to Zipkin.

this, and to a lesser degree the hot partitions, are the hurdles to making this a million+ writes per second backend (even though i can't see anymore provisioning hundreds of virtual server c* nodes just for the sake of zipkin visibility…)

in benchmarking i was able to write comfortably 150 spans per second per cpu, over any hours, keeping latency generally in double-digit milliseconds. (cpu isn't really the best scaling base, but for typical cpu-ram-diskIO, it's good enough. this benchmarking was on a laptop). 
when i dropped just the one SASI on annotation_query, I could then write comfortably 6k spans per second per cpu, over many hours, keeping latency in single-digit milliseconds.